File: core/src/main/java/com/sequenceiq/cloudbreak/converter/JsonToFailurePolicyConverter.java
package com.sequenceiq.cloudbreak.converter;

import org.springframework.stereotype.Component;

import com.sequenceiq.cloudbreak.api.model.FailurePolicyRequest;
import com.sequenceiq.cloudbreak.domain.FailurePolicy;

@Component
public class JsonToFailurePolicyConverter extends AbstractConversionServiceAwareConverter<FailurePolicyRequest, FailurePolicy> {
    @Override
    public FailurePolicy convert(FailurePolicyRequest json) {
        FailurePolicy stackFailurePolicy = new FailurePolicy();
        stackFailurePolicy.setAdjustmentType(json.getAdjustmentType());
        stackFailurePolicy.setThreshold(json.getThreshold() == null ? 0 : json.getThreshold());
        return stackFailurePolicy;
    }
}

File: core/src/main/java/com/sequenceiq/cloudbreak/converter/JsonToFailurePolicyConverter.kt
package com.sequenceiq.cloudbreak.converter

import org.springframework.stereotype.Component

import com.sequenceiq.cloudbreak.api.model.FailurePolicyJson
import com.sequenceiq.cloudbreak.domain.FailurePolicy

@Component
class JsonToFailurePolicyConverter : AbstractConversionServiceAwareConverter<FailurePolicyJson, FailurePolicy>() {
    override fun convert(json: FailurePolicyJson): FailurePolicy {
        val stackFailurePolicy = FailurePolicy()
        stackFailurePolicy.adjustmentType = json.adjustmentType
        stackFailurePolicy.id = json.id
        stackFailurePolicy.threshold = if (json.threshold == null) 0 else json.threshold
        return stackFailurePolicy
    }
}
